r/R/install-arrow.R

```diff
@@ -0,0 +1,137 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#' Help installing the Arrow C++ library
+#'
+#' Binary package installations should come with a working Arrow C++ library,
+#' but when installing from source, you'll need to obtain the C++ library
+#' first. This function offers guidance on how to get the C++ library depending
+#' on your operating system and package version.
+#' @export
+#' @importFrom utils packageVersion
+install_arrow <- function() {
+  os <- tolower(Sys.info()[["sysname"]])
+  # c("windows", "darwin", "linux", "sunos") # win/mac/linux/solaris
+  version <- packageVersion("arrow")
+  # From CRAN check:
+  rep <- installed.packages(fields="Repository")["arrow", "Repository"]
+  from_cran <- identical(rep, "CRAN")
+  # Is it possible to tell if was a binary install from CRAN vs. source?
+
+  message(install_arrow_msg(arrow_available(), version, from_cran, os))
+}
+
+install_arrow_msg <- function(has_arrow, version, from_cran, os) {
+  # TODO: check if there is a newer version on CRAN?
+
+  # install_arrow() sends "version" as a "package_version" class, but for
+  # convenience, this also accepts a string like "0.13.0". Calling
+  # `package_version` is idempotent so do it again, and then `unclass` to get
+  # the integers. Then see how many there are.
+  dev_version <- length(unclass(package_version(version))[[1]]) > 3
+  # Based on these parameters, assemble a string with installation advice
+  if(has_arrow) {
```

Comment:
Can you have a look at https://style.tidyverse.org/syntax.html#indenting esp space after `if`. We don't enforce here, perhaps https://github.com/r-lib/styler can help

Reply:
Certainly. I was trying to follow what I inferred the code style to be based on what I saw in the package already, but now that I grep for `if(`, I don't see much, so I think I just spent too much time in `tools/winlibs.R` last week 😅

While I'm at it, I'll find/replace the other bad-style ifs since there aren't many. 

I'm generally in favor of automation but I don't think we need `styler` yet. If/when we get more contributions from the community, it may be worth it then.